components/campaign/CampaignChaosBagView/index.tsx

```diff
@@ -0,0 +1,363 @@
+    if (buttonId === 'back' || buttonId === 'androidBack') {
+      Navigation.pop(componentId);
+    } else if (buttonId === 'edit') {
+      this._showChaosBagDialog();
+    }
+  }
+
+  _handleClearTokensPressed = () => {
+    this.clearTokens();
+  };
+
+  _handleDrawTokenPressed = () => {
+    const { chaosBagResults } = this.props;
+    if (chaosBagResults.drawnTokens.length >= 1) {
+      this.clearTokens();
+    } else {
+      this.drawToken();
+    }
+  };
+
+  _handleAddAndDrawAgainPressed = () => {
+    this.drawToken();
+  };
+
+  _handleSealTokenPressed = () => {
+  };
+
+  _showChaosBagDialog = () => {
+    const {
+      componentId,
+      chaosBag,
+      updateChaosBag,
+    } = this.props;
+
+
+    Navigation.push<EditChaosBagProps>(componentId, {
+      component: {
+        name: 'Dialog.EditChaosBag',
+        passProps: {
+          chaosBag,
+          updateChaosBag: updateChaosBag,
+          trackDeltas: true,
+        },
+        options: {
+          topBar: {
+            title: {
+              text: t`Chaos Bag`,
+            },
+            backButton: {
+              title: t`Cancel`,
+            },
+          },
+        },
+      },
+    });
+  };
+
+  getWeighedChaosBag = (list: any[], weight: any[]) => {
+    const weighedList = [];
+
+    for (let i = 0; i < weight.length; i++) {
+      const multiples = weight[i];
+
+      for (let j = 0; j < multiples; j++) {
+        weighedList.push(list[i]);
+      }
+    }
+
+    return weighedList;
+  }
+
+  getRandomChaosToken = (chaosBag: ChaosBag) => {
+    const list = keys(chaosBag);
+    const weight = values(chaosBag);
+    const weighedList = this.getWeighedChaosBag(list, weight);
+
+    return shuffle(weighedList)[0];
+  };
+
+  drawToken = () => {
+    const { campaignId, chaosBag, chaosBagResults } = this.props;
+
+    const newIconKey = this.getRandomChaosToken(chaosBag);
+    const drawnTokens = chaosBagResults.drawnTokens;
+
+    if (newIconKey) {
+      drawnTokens.push(newIconKey);
+    }
+
+    const newChaosBagResults = {
+      drawnTokens: drawnTokens,
+      sealedTokens: chaosBagResults.sealedTokens,
+      totalDrawnTokens: chaosBagResults.totalDrawnTokens + 1,
+    };
+
+    updateCampaign(campaignId, { chaosBagResults: newChaosBagResults });
+  }
+
+  // _mutateCount = (id: ChaosTokenType, mutate: (count: number) => number) => {
+  //   this.setState((state: State) => {
+  //     const newChaosBag = Object.assign(
+  //       {},
+  //       state.chaosBag,
+  //       { [id]: mutate(state.chaosBag[id] || 0) }
+  //     );
+  //     return {
+  //       chaosBag: newChaosBag,
+  //     };
+  //   });
+  // };
+
+  clearTokens = () => {
+    const { campaignId, chaosBagResults } = this.props;
+
+    const newChaosBagResults = {
+      drawnTokens: [],
+      sealedTokens: chaosBagResults.sealedTokens,
+      totalDrawnTokens: chaosBagResults.totalDrawnTokens,
+    };
+
+    updateCampaign(campaignId, { chaosBagResults: newChaosBagResults });
```

Comment:
Actually, second problem.  You need to use the updateCampaign that comes out of props, not the action that's imported.  I've made this mistake SO many times myself.